packages/cli/src/decorators/registerController.ts

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { Router } from 'express';
+import type { Application, Request, RequestHandler as Middleware, Response } from 'express';
+import { BASE_PATH, Controller, RouteMetadata, ROUTES } from './types';
+import { send } from '@/ResponseHelper'; // TODO: move `ResponseHelper.send` to this file
+import type { Config } from '@/config';
+
+export const registerController = (
+	app: Application,
+	config: Config,
+	instance: object,
+	...middlewares: Middleware[]
+): void => {
+	const controllerClass = instance.constructor;
+	const routes = Reflect.getMetadata(ROUTES, controllerClass) as RouteMetadata[];
```

Comment:
this will be undefined if we add a new controller with no routes, in which case the server will fail to start. we can add an explicit error, but I don't think we need to right now.